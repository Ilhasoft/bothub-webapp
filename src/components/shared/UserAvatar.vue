<template>
  <button
    type="button"
    class="avatar">
    <b-icon
      icon="account"
      class="avatar-icon" />
  </button>
</template>

<script>
export default {
  name: 'UserAvatar',
  props: {
    profile: {
      required: true,
      type: Object,
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~@/assets/scss/utilities.scss';

.avatar {
  $size: 36px;

  position: relative;
  width: $size;
  height: $size;
  overflow: hidden;
  border-radius: 100%;
  border: 2px solid #fff;
  background-color: $white-ter;
  outline: none;
  cursor: pointer;

  &-icon {
    position: absolute;
    left: 50%;
    height: 50%;
    transform: translate(-50%, -50%);
    color: $grey-light;
  }
}
</style>
