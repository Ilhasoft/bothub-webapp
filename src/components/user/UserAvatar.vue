<template>
  <div
    :class="{
      'avatar': true,
      [`avatar--${size}`]: !!size,
      'avatar--clickable': clickable,
    }"
    @click="$emit('click', $event)"
  >
    <div
      :class="{avatar__content: true,
               [`avatar--${size}`]: !!size}">
      <img
        v-if="!isOrganization && profile.nickname"
        :class="size"
        :src="`https://robohash.org/${profile.nickname}`">
      <div
        v-else
        class="avatar__icon">
        <b-icon
          :icon="isOrganization ? 'account-multiple' : 'account'"
          size="small" />
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'UserAvatar',
  props: {
    profile: {
      required: true,
      type: Object,
    },
    isOrganization: {
      type: Boolean,
      default: false,
    },
    size: {
      type: String,
      default: 'normal',
    },
    clickable: {
      type: Boolean,
      default: true,
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~@/assets/scss/variables.scss';
@import '~@/assets/scss/colors.scss';

  .avatar {
    position: relative;
    width: $size-normal;
    height: $size-normal;
    overflow: hidden;
    color: $color-fake-black;
    background-color: $color-fake-white;
    border-color: transparent;
    border-radius: 50%;
    outline: none;
    display: inline-block;

    @each $size in $sizes {
      $size-name: nth($size, 1);
      $size-value: nth($size, 2);

      &--#{$size-name} {
        width: $size-value;
        height: $size-value;
      }

      &--full {
        width: 100%;
        height: 100%;
      }
    }

    &__icon {
      width: 100%;
      height: 100%;
      display: flex;
      justify-content: center;
      align-items: center;
    }

    &--clickable {
      cursor: pointer;
    }

    &__content {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
    }
  }
</style>
