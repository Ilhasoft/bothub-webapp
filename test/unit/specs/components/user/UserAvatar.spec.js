import BH from 'bh';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import UserAvatar from '@/components/user/UserAvatar';


const localVue = createLocalVue();
localVue.use(BH);

describe('UserAvatar.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(UserAvatar, {
      localVue,
      propsData: { profile: {} },
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('with small size', () => {
    beforeEach(() => {
      wrapper.setProps({ size: 'small' });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
