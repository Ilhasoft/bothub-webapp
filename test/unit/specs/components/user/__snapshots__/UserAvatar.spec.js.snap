// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar.vue renders correctly 1`] = `
<div class="avatar avatar--clickable avatar--normal">
  <div class="avatar__content avatar--normal"><img src="https://robohash.org/botinho" class="normal"></div>
</div>
`;

exports[`UserAvatar.vue with small size renders correctly 1`] = `
<div class="avatar avatar--clickable avatar--small">
  <div class="avatar__content avatar--small"><img src="https://robohash.org/botinho" class="small"></div>
</div>
`;
